### PR TITLE
Update ldapserver

### LINKED ISSUE #1: does not shut down cleanly
the ldap servers Stop() needs to be called on context.Cancel() ... or we make the ldap serve cance l aware. Going for the former.

### PR DESCRIPTION
**Pull request by butonic**
fixes https://github.com/owncloud/ocis-devldap/issues/1

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@butonic drone hates you :dancing_men:

**Review by DeepDiver1975 — APPROVED** [review]

File: pkg/assets/assets.go [hunk 1]
```diff
@@ -16,35 +17,41 @@ type assets struct {
...
-		content, err := ioutil.ReadFile(a.config.LDAP.Data)
-
-		if err != nil {
-			a.logger.Error().
-				Str("path", a.config.LDAP.Data).
-				Msg("Failed to read data file")
-
-			return []byte(), err
+// Open just implements the HTTP filesystem interface.
+func (a assets) Open(original string) (http.File, error) {
```
Comment (tboerger) [inline pkg/assets/assets.go:21]: Having this http file handler doesn't make sense at all for this extensions. Why have you replaced the Data function?

File: pkg/command/server.go [hunk 2]
```diff
@@ -118,7 +117,7 @@ func Server(cfg *config.Config) cli.Command {
 			var (
 				gr          = run.Group{}
 				ctx, cancel = context.WithCancel(context.Background())
-				metrics     = metrics.New()
+				//metrics     = metrics.New()
```
Comment (tboerger) [inline pkg/command/server.go:120]: You should not remove that, you should feed that with metrics!

File: pkg/config/config.go [hunk 3]
```diff
@@ -30,13 +30,19 @@ type Tracing struct {
 	Service   string
 }
 
+// Asset defines the available asset configuration.
+type Asset struct {
```
Comment (tboerger) [inline pkg/config/config.go:34]: Same for this struct, there are no assets, there is only the data path.

**Pull request merged**